fuzz/fuzz_targets/fuzz_parse_sql.rs

```diff
@@ -0,0 +1,12 @@
+use honggfuzz::fuzz;
+use sqlparser::dialect::GenericDialect;
+use sqlparser::parser::Parser;
+
+fn main() {
+    loop {
+        fuzz!(|data: String| {
```

Comment:
This is pretty cool. Where does the data come from? Is it just random strings? Or does this somehow know how to make valid sql statements?

Reply:
I see some related comments here: https://github.com/sqlparser-rs/sqlparser-rs/pull/211#issuecomment-650716166

Reply:
For now it is some random bytes, but can be used to find bugs. Optimization can be continued later, refer to: https://www.cockroachlabs.com/blog/sqlsmith-randomized-sql-testing/

Reply:
In a past project, we had a harness that could generate random SQL queries and it found **many** bugs -- such tests are a wonderful way to help database software mature

Reply:
Filed https://github.com/apache/arrow-datafusion/issues/913 for DataFusion. Thanks for the pointer @PsiACE